.circleci/scripts/upload_cloud_watch_to_s3.py

```diff
@@ -2,22 +2,42 @@
 import time
 import sys
 
+import botocore
 import boto3
 
 logs = boto3.client("logs")
 
 
-def upload_cloud_watch_logs(log_group_name: str, bucket_name: str, prefix: str) -> str:
-    task_id = logs.create_export_task(
-        logGroupName=log_group_name,
-        fromTime=0,
-        to=int(round(time.time() * 1000)),
-        destination=bucket_name,
-        destinationPrefix=prefix
-    )["taskId"]
+def upload_cloud_watch_logs(log_group_name: str, bucket_name: str, prefix: str, retries: int) -> str:
+    attempts = 0
+
+    while attempts <= retries:
+        try:
+            task_id = logs.create_export_task(
+                logGroupName=log_group_name,
+                fromTime=0,
+                to=int(round(time.time() * 1000)),
+                destination=bucket_name,
+                destinationPrefix=prefix
+            )["taskId"]
+
+            print(f"Uploading logs to s3://{bucket_name}/{prefix}/{task_id}.")
+            return task_id
+
+        except botocore.exceptions.ClientError as error:
+            if error.response['Error']['Code'] == 'LimitExceededException':
+                print(f"Got an error possibly due to concurrent uploads: {error}.")
+            else:
+                raise error
 
-    print(f"Uploading logs to s3://{bucket_name}/{prefix}/{task_id}.")
-    return task_id
+            attempts += 1
+            if attempts > retries:
+                raise error
+
+            time.sleep(3)
```

Comment:
I get the feeling we should probably sleep a little longer. Do you know how long the export task hangs around before its done?

We see this so often, I feel if a 3 second window was sufficient I'd be that would be a little surprised. We could do `time.sleep(5 ** attempts)` or just make it like 30 seconds. Unless you've already tested it in which case 👌 .

Reply:
I tested it and export tasks seems to take around a second, but will bump this up to 10 sec just in case.